packages/web3/test/e2e/get_block_transaction_count.test.ts

```diff
@@ -0,0 +1,107 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { isBigInt, isHexStrict, isNumber, isString } from 'web3-validator';
+
+import Web3, { FMT_BYTES, FMT_NUMBER } from '../../src';
+import { getSystemE2ETestProvider } from './e2e_utils';
+import { closeOpenConnection, getSystemTestBackend } from '../shared_fixtures/system_tests_utils';
+import { toAllVariants } from '../shared_fixtures/utils';
+
+describe(`${getSystemTestBackend()} tests - getBlockTransactionCount`, () => {
+	const provider = getSystemE2ETestProvider();
+	const blockData: {
+		earliest: 'earliest';
+		latest: 'latest';
+		pending: 'pending';
+		finalized: 'finalized';
+		safe: 'safe';
+		blockNumber: number;
+		blockHash: string;
+	} = {
+		earliest: 'earliest',
+		latest: 'latest',
+		pending: 'pending',
+		finalized: 'finalized',
+		safe: 'safe',
+		blockNumber: getSystemTestBackend() === 'sepolia' ? 3240768 : 17029884,
+		blockHash:
+			getSystemTestBackend() === 'sepolia'
+				? '0xe5e66eab79bf9236eface52c33ecdbad381069e533dc70e3f54e2f7727b5f6ca'
+				: '0x2850e4a813762b2de589fa5268eacb92572defaf9520608deb129699e504cab2',
+	};
+
+	let web3: Web3;
+
+	beforeAll(() => {
+		web3 = new Web3(provider);
+	});
+
+	afterAll(async () => {
+		await closeOpenConnection(web3);
+	});
+
+	it.each(
+		toAllVariants<{
+			block:
+				| 'earliest'
+				| 'latest'
+				| 'pending'
+				| 'finalized'
+				| 'safe'
+				| 'blockHash'
+				| 'blockNumber';
+			format: string;
+		}>({
+			block: [
+				'earliest',
+				'latest',
+				'pending',
+				'safe',
+				'finalized',
+				'blockHash',
+				'blockNumber',
+			],
+			format: Object.values(FMT_NUMBER),
+		}),
+	)('getBlockTransactionCount', async ({ block, format }) => {
+		const result = await web3.eth.getBlockTransactionCount(blockData[block], {
+			number: format as FMT_NUMBER,
+			bytes: FMT_BYTES.HEX,
+		});
+
+		switch (format) {
+			case 'NUMBER_NUMBER':
+				// eslint-disable-next-line jest/no-conditional-expect
+				expect(isNumber(result)).toBeTruthy();
+				break;
+			case 'NUMBER_HEX':
+				// eslint-disable-next-line jest/no-conditional-expect
+				expect(isHexStrict(result)).toBeTruthy();
```

Comment:
For most of existing tests , I'll suggest to add some definite test cases as well for validating that we are getting and processing correctly in our lib. Like for getting block transaction count instead of :
`expect(isString(result)).toBeTruthy();`
there should be test of getting number of transaction for some specific block and validating that on mainnet or sp nw. 

And same for other eth functions in other test files where applicable. Thanks @spacesailor24 for your efforts.

Reply:
@jdevcs This is hard to do when using block tags because anything other than `earliest` will return a recent block where the properties are non-deterministic for our tests e.g. we won't know what the block transaction count will be for the `latest` block, we won't know the block number, hash, etc. of the `latest` block, and we won't know the balance of the account for these block tags

Reply:
We should be able to check exact properties when we make the request using a block hash or number though, so I'll update the tests to do that

Reply:
You are right, Thats why I requested to :

> there should be test of getting number of transaction for some `specific block` and validating that on mainnet or sp nw.

Reply:
@jdevcs Updated via [this](https://github.com/web3/web3.js/pull/5990/commits/dfd111793b4dab26eee936567e84a3f8e3d7965b) commit